src/main/java/reposense/parser/DateArgumentType.java

```diff
@@ -1,50 +1,24 @@
 package reposense.parser;
 
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
 import java.util.Date;
 import java.util.Optional;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 
 import net.sourceforge.argparse4j.inf.Argument;
 import net.sourceforge.argparse4j.inf.ArgumentParser;
 import net.sourceforge.argparse4j.inf.ArgumentParserException;
 import net.sourceforge.argparse4j.inf.ArgumentType;
+import reposense.util.TimeUtil;
 
 /**
  * Verifies and parses a string-formatted date to a {@code Date} object.
  */
 public class DateArgumentType implements ArgumentType<Optional<Date>> {
-    protected static final String DATE_FORMAT_REGEX =
-            "^((0[1-9]|[12][0-9]|3[01])\\/(0[1-9]|1[012])\\/(19|2[0-9])[0-9]{2})";
     private static final String PARSE_EXCEPTION_MESSAGE_INVALID_DATE_STRING_FORMAT = "Invalid Date: %s";
-    private static final DateFormat CLI_ARGS_DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss");
-
-    static {
-        /*
-         * Setting setLenient to false prevents unexpected results.
-         * Without it, even with "dd/MM/yyyy HH:mm:ss" format, 11/31/2017 00:00:00 will be parsed to 11/7/2019 00:00:00.
-         * */
-        CLI_ARGS_DATE_FORMAT.setLenient(false);
-    }
-
-    /**
-     * Extracts the first substring that matches the {@code DATE_FORMAT_REGEX}.
-     */
-    protected String extractDate(String date) {
-        Matcher matcher = Pattern.compile(DATE_FORMAT_REGEX).matcher(date);
-        String extractedDate = date;
-        if (matcher.find()) {
-            extractedDate = matcher.group(1);
-        }
-        return extractedDate;
-    }
```

Comment:
Don't really agree with moving this code into TimeUtil since the date extraction seems like a parsing thing. Also this file is already quite short. TimeUtil seems to be for stuff related to Date objects, but this takes in a string and outputs a string.

Reply:
This PR needs to be viewed in the context of #1501. The date parsing is also needed in report-config.json, which is why it is extracted out to this file.

Also, this file contains time-based functionalities, not Time-Object functionalities or Date-Object functionalities. Hence, it returns a string.

Reply:
Based on the logic I feel it's okay to extract it out to a different file otherwise there will be a lot of code duplication. Though it seems like a parsing thing, it is better to not have duplicate code in the codebase if it can be avoided. Therefore, in my opinion, I think this refactoring and moving is okay. 
@0blivious your opinion?